### ISSUE TITLE
Pyright 383 regression - reportArgumentType/npt.ArrayLike

### ISSUE THREAD

**Issue by danielpopescu** [microsoft/pyright#9132]
```python

from __future__ import annotations

from typing import Any

import numpy as np
import numpy.typing as npt


def as_array(a: npt.ArrayLike) -> npt.NDArray[Any]:
    return np.array(a)


def test() -> int:
    lst = as_array([[1, 2], [3, 4]])
    return len(lst)

```

there is a new type error  on the line lst is defined.

This worked in 382. Python 3.12.x, numpy 2.1.1

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.384

**Comment by danielpopescu** [comment]
agreed. thank you
